src/vma/sock/sockinfo_udp.cpp

```diff
@@ -1787,6 +1787,17 @@ bool sockinfo_udp::rx_input_cb(mem_buf_desc_t* p_desc, void* pv_fd_ready_array)
 		return false;
 	}
 
+	if ((m_mc_num_grp_with_src_filter) && (IN_MULTICAST_N(p_desc->path.rx.dst.sin_addr.s_addr))) {
+		in_addr_t mc_grp = p_desc->path.rx.dst.sin_addr.s_addr;
+		in_addr_t mc_src = p_desc->path.rx.src.sin_addr.s_addr;
+
+		if ((m_mc_memberships_map.find(mc_grp) == m_mc_memberships_map.end()) ||
+			((0 < m_mc_memberships_map[mc_grp].size()) &&
+			(m_mc_memberships_map[mc_grp].find(mc_src) == m_mc_memberships_map[mc_grp].end()))) {
+				return false;
```

Comment:
please add 'discard' log message similar to other test cases: `si_udp_logfunc("rx packet discarded - multicast source mismatch ...");`